### PR TITLE
Use Error throwing instead of is_raw_result_request option

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Use Error throwing instead of is_raw_result_request option
- Update tests for latest json rpc API result format changes

Related PRs:
- https://github.com/ainblockchain/ain-blockchain/pull/1070

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/1053

### CODE REVIEW

File: src/provider.ts [hunk 1]
```diff
@@ -41,11 +41,10 @@ export default class Provider {
     };
     const response = await this.httpClient.post(this.apiEndpoint, data);
     const rawResult = get(response, 'data.result');
-    // TODO(platfowner): Remove this block once migration is completed.
-    if (!data.params.is_raw_result_request) {
-      return rawResult.code !== undefined ? rawResult : get(rawResult, 'result', null);
+    if (typeof rawResult !== 'object' || !(rawResult.code === undefined || rawResult.code === 0)) {
+      throw new Error(JSON.stringify(rawResult));
```
Comment (liayoo) [inline src/provider.ts:45]: Could you consider defining an error class that extends Error? 
ref: https://github.com/ainblockchain/ain-blockchain/blob/master/consensus/index.js#L33

File: __tests__/ain.test.ts [hunk 2]
```diff
@@ -360,18 +360,21 @@ describe('ain-js', function() {
...
+        "result": true,
         "code": 0,
+        "protoVer": "erased",
       });
     });
 
+    it('validateAppName returns false', async function () {
+      await expect(() => ain.validateAppName('app/path'))
+          .rejects
+          .toThrow(/"is_valid":false,"result":false,"code":30601,"message":"Invalid app name for state label: app\/path"/);
```
Comment (minsulee2) [inline __tests__/ain.test.ts:375]: 🙌

File: src/types.ts [hunk 3]
```diff
@@ -59,14 +59,12 @@ export type GetOperationType = "GET_VALUE" | "GET_RULE" | "GET_OWNER" | "GET_FUN
...
 export type GetOptions = {
-  is_global?: boolean,
-  is_final?: boolean,
-  is_shallow?: boolean,
-  // TODO(platfowner): Remove this option once migration is completed.
-  is_raw_result_request?: boolean,
-  include_version?: boolean,
-  include_tree_info?: boolean,
-  include_proof?: boolean,
+  is_global?: boolean;
```
Comment (minsulee2) [inline src/types.ts:62]: Q) I am quite new in typescript. What makes this difference? I mean , and ;

**Review by minsulee2 — APPROVED** [review]
LGTM with minor question

File: src/types.ts (hunk 3 above)
Comment (platfowner) [reply src/types.ts:62]: It looks like ';' is more common for types and interfaces although ',' doesn't cause compile errors.

File: src/provider.ts (hunk 1 above)
Comment (platfowner) [reply src/provider.ts]: That's a good idea. Just done. PTAL

**Comment by platfowner** [comment]
Comments addressed. PTAL @liayoo @minsulee2

File: src/errors.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export class BlockchainError extends Error {
+  public code: number;
+
+  constructor(code: number, message: string) {
+    super(message);
+    // NOTE(platfowner): https://stackoverflow.com/questions/68899615/how-to-expect-a-custom-error-to-be-thrown-with-jest
```
Comment (liayoo) [inline src/errors.ts:6]: 👍

**Review by liayoo — APPROVED** [review]
LGTM, thanks!!

**Comment by platfowner** [comment]
Thanks for the review. Now merging.. @liayoo @minsulee2

**Pull request merged**